hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueProxySupport.java

```diff
@@ -175,12 +167,16 @@ protected void checkObjectNotNull(Object o) {
     }
 
     private <T> T invokeAndGet(QueueOperation operation) {
+        return invokeAndGet(operation, RuntimeException.class);
```

Comment:
I can't tell I particularly like this. From the other hand I'm not sure how to solve it more elegantly.